onnxruntime/test/providers/vitisai/vitisai_basic_test.cc

```diff
@@ -0,0 +1,316 @@
+   Struct res_int8_10k = VAIExecutionProviderTest<int8_t>(int8_10k, 10000);
+   Struct res_fp32_100k = VAIExecutionProviderTest<float>(fp32_100k, 100000);
+   Struct res_int8_100k = VAIExecutionProviderTest<int8_t>(int8_100k, 100000);
+   Struct res_fp32_1m = VAIExecutionProviderTest<float>(fp32_1m, 1000000);
+   Struct res_int8_1m = VAIExecutionProviderTest<int8_t>(int8_1m, 1000000);
+   Struct res_fp32_10m = VAIExecutionProviderTest<float>(fp32_10m, 10000000);
+   Struct res_int8_10m = VAIExecutionProviderTest<int8_t>(int8_10m, 10000000);
+
+   delete[] int8_10k;
+   delete[] int8_100k;
+   delete[] int8_1m;
+   delete[] int8_10m;
+   delete[] fp32_10k;
+   delete[] fp32_100k;
+   delete[] fp32_1m;
+   delete[] fp32_10m;
+
+  system("wmic path win32_VideoController get name > gpu_info.txt");
+  system("wmic cpu get name > sys_info.txt");
+  system("wmic memorychip get speed >> sys_info.txt");
+
+  std::ifstream file("sys_info.txt");
+  std::string str;
+  std::string GPUName;
+  std::string CPUName;
+  std::string memorySpeed;
+
+  int count = 0;
+  while (std::getline(file, str)) {
+   if (count == 1) {
+      CPUName = str;
+    } else if (count == 3) {
+      memorySpeed = str;
+    }
+    count++;
+  }  
+
+  std::ifstream file_gpu("gpu_info.txt");
+  count = 0;
+  while (std::getline(file_gpu, str)) {
+    if (count == 2) {
+      GPUName = str;
+      break;
+    } 
+    count++;
+  }  
+
+  std::ofstream myfile;
+  myfile.open("results.csv");
+  
+  myfile << "System information: \n";
+  myfile << "CPU:," << CPUName;
+  myfile << "GPU:," << GPUName;
+  myfile << "Memory speed [MHz]:," << memorySpeed;
+  myfile << "Bus speed [MHz]:,"
+         << "99.98 \n";
+  myfile << "\n";
+
+  myfile << "2D Performance (RGBA): \n";
+  myfile << "tensor size,GPU->CPU [us], CPU->GPU [us] \n";
+  myfile << "1920x1080 ~8M," << res_1920x1080.readback << "," << res_1920x1080.upload << ",\n ";
+  myfile << "2560x1440 ~15M," << res_2560x1440.readback << "," << res_2560x1440.upload << ",\n ";
+  myfile << "3840x1260 ~19M," << res_3840x1260.readback << "," << res_3840x1260.upload << ",\n ";
+  myfile << "\n";
+
+  myfile << "1D Performance: \n";
+  myfile << "tensor size, int8 (GPU->CPU) [us], int8 (CPU->GPU) [us],fp32 (GPU->CPU) [us], fp32 (CPU->GPU) [us] \n";
+  myfile << "10K," << res_int8_10k.readback << "," << res_int8_10k.upload << "," << res_fp32_10k.readback << "," << res_fp32_10k.upload
+         << ",\n ";
+  myfile << "100K," << res_int8_100k.readback << "," << res_int8_100k.upload << "," << res_fp32_100k.readback << "," << res_fp32_100k.upload
+         << ",\n ";
+  myfile << "1M," << res_int8_1m.readback << "," << res_int8_1m.upload << "," << res_fp32_1m.readback << "," << res_fp32_1m.upload
+         << ",\n ";
+  myfile << "10M," << res_int8_10m.readback << "," << res_int8_10m.upload << "," << res_fp32_10m.readback << "," << res_fp32_10m.upload
+         << ",\n ";
+  myfile.close();
```

Comment:
Not required, the function is called by the destructor.
https://en.cppreference.com/w/cpp/io/basic_ofstream/close